File: Selenium/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:

    wait = WebDriverWait(driver, 10)

    driver.get("https://training-support.net/webelements/dynamic-content")

    print("Page title:", driver.title)

    driver.find_element(By.ID, "genButton").click()

    wait.until(EC.text_to_be_present_in_element((By.ID, "word"), "release"))

    word = driver.find_element(By.ID, "word").text
    print("Word found:", word)

File: Selenium/Activity9.java
package examples;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity9 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-content");

        System.out.println("Page title: " + driver.getTitle());

        driver.findElement(By.id("genButton")).click();

        wait.until(ExpectedConditions.textToBePresentInElementLocated(
                By.id("word"), "release"));

        String word = driver.findElement(By.id("word")).getText();
        System.out.println("Word found: " + word);

        driver.quit();
    }
}